ui/app/AppLayouts/Chat/ChatColumn/ChatComponents/SignTransactionModal.qml

```diff
@@ -0,0 +1,190 @@
+
+            GasSelector {
+                id: gasSelector
+                anchors.topMargin: Style.current.bigPadding
+                slowestGasPrice: parseFloat(walletModel.safeLowGasPrice)
+                fastestGasPrice: parseFloat(walletModel.fastestGasPrice)
+                getGasEthValue: walletModel.getGasEthValue
+                getFiatValue: walletModel.getFiatValue
+                defaultCurrency: walletModel.defaultCurrency
+                width: stack.width
+                reset: function() {
+                    slowestGasPrice = Qt.binding(function(){ return parseFloat(walletModel.safeLowGasPrice) })
+                    fastestGasPrice = Qt.binding(function(){ return parseFloat(walletModel.fastestGasPrice) })
+                }
+            }
+        }
+        TransactionFormGroup {
+            id: group2
+            headerText: qsTr("Transaction preview")
+            footerText: qsTr("Sign with password")
+
+            TransactionPreview {
+                id: pvwTransaction
+                width: stack.width
+                fromAccount: root.selectedAccount
+                gas: {
+                    const value = walletModel.getGasEthValue(gasSelector.selectedGasPrice, gasSelector.selectedGasLimit)
+                    const fiatValue = walletModel.getFiatValue(value, "ETH", walletModel.defaultCurrency)
+                    return { value, "symbol": "ETH", fiatValue }
+                }
+                toAccount: root.selectedRecipient
+                asset: root.selectedAsset
+                amount: { "value": root.selectedAmount, "fiatValue": root.selectedFiatAmount }
+                currency: walletModel.defaultCurrency
+                reset: function() {}
+            }
+        }
+        TransactionFormGroup {
+            id: group3
+            headerText: qsTr("Sign with password")
+            footerText: qsTr("Send %1 %2").arg(root.selectedAmount).arg(!!root.selectedAsset ? root.selectedAsset.symbol : "")
+
+            TransactionSigner {
+                id: transactionSigner
+                width: stack.width
+                signingPhrase: walletModel.signingPhrase
+                reset: function() {
+                    signingPhrase = Qt.binding(function() { return walletModel.signingPhrase })
+                }
+            }
+        }
+    }
+
+    footer: Item {
+        anchors.top: parent.top
+        anchors.left: parent.left
+        anchors.right: parent.right
+        StyledButton {
+            id: btnBack
+            anchors.left: parent.left
+            width: 44
+            height: 44
+            visible: !stack.isFirstGroup
+            label: ""
+            background: Rectangle {
+                anchors.fill: parent
+                border.width: 0
+                radius: width / 2
+                color: btnBack.disabled ? Style.current.grey :
+                        btnBack.hovered ? Qt.darker(btnBack.btnColor, 1.1) : btnBack.btnColor
```

Comment:
At what point is this button disabled? Maybe this isn't needed? Also, do we have a dark mode version of `grey` (IOW, maybe we could create/use `Style.current.disabled` instead)?